lib/site_settings/deprecated_settings.rb

```diff
@@ -35,6 +35,64 @@ module SiteSettings::DeprecatedSettings
     ["min_trust_level_to_allow_ignore", "ignore_allowed_groups", false, "3.3"],
   ]
 
+  OVERRIDE_TL_GROUP_SETTINGS = %w[
+    anonymous_posting_min_trust_level
+    shared_drafts_min_trust_level
+    min_trust_level_for_here_mention
+    approve_unless_trust_level
+    approve_new_topics_unless_trust_level
+    email_in_min_trust
+    min_trust_to_edit_wiki_post
+    allow_uploaded_avatars
+    min_trust_to_create_topic
+    min_trust_to_edit_post
+    min_trust_to_flag_posts
+    tl4_delete_posts_and_topics
+    min_trust_level_to_allow_user_card_background
+    min_trust_level_to_allow_invite
+    min_trust_level_to_allow_ignore
+  ]
+
+  def group_to_tl(old_setting, new_setting)
+    tl_and_staff =
+      SiteSetting.type_supervisor.get_type(old_setting.to_sym) == :enum &&
+        SiteSetting
+          .type_supervisor
+          .get_enum_class(old_setting.to_sym)
+          .is_a?(TrustLevelAndStaffSetting)
+
+    min_trust_from_new_group_values =
+      self
+        .public_send("#{new_setting}_map")
+        .select do |group_id|
+          # We only want auto groups, no actual groups because they cannot be
+          # mapped to TLs.
+          group_id.between?(
+            tl_and_staff ? Group::AUTO_GROUPS[:admins] : Group::AUTO_GROUPS[:trust_level_0],
+            Group::AUTO_GROUPS[:trust_level_4],
+          )
+        end
+        .min
+
+    if min_trust_from_new_group_values == Group::AUTO_GROUPS[:admin]
```

Comment:
Consider the case where the new setting has: `1|3|13` (admins, staff, and TL3).

In this case taking the minimum one would "miss" the TL3?

Reply:
Ah yes you are right, will fix